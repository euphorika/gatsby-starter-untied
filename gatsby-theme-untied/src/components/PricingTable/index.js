/** @jsx jsx */
import { jsx } from 'theme-ui'
import PropTypes from 'prop-types'

import { Link } from 'gatsby'
import Img from 'gatsby-image'

const PricingTable = ({
  children,
  id,
  headline,
  price,
  imgFluid,
  callToAction,
}) => (
  <section
    id={id}
    sx={{
      mx: ['auto', 3],
      my: 4,
      width: '100%',
      maxWidth: '300px',
      fontSize: 0,
      py: 2,
      border: 'solid 1px',
      borderColor: 'muted',
      scrollSnapAlign: 'start',
      flexShrink: ['0', 'unset'],
      mr: ['50px', '15px'],
      webkitTransformOrigin: 'center center',
      transformOrigin: 'center center',
      webkitTransform: 'scale(1)',
      transform: 'scale(1)',
      webbkitTransition: '-webkit-transform 0.5s',
      transition: 'transform 0.5s',
    }}
  >
    <h3
      sx={{
        m: 0,
        px: 3,
        fontSize: 2,
      }}
    >
      {headline}
    </h3>
    <p
      sx={{
        mb: 2,
        px: 3,
        fontSize: 2,
      }}
    >
      {price}
    </p>
    <Img fluid={imgFluid} alt={headline} />
    <div
      sx={{
        mt: 2,
        px: 3,
      }}
    >
      {children}
    </div>
    <div
      sx={{
        mt: 3,
        px: 3,
      }}
    >
      <Link
        sx={{
          display: 'inline-block',
          width: '100%',
          textAlign: 'center',
          color: 'btnText',
          fontSize: 0,
          fontWeight: 'bold',
          bg: 'btnBg',
          textDecoration: 'none',
          borderRadius: 0,
          p: 2,
        }}
        to={callToAction.link}
      >
        {callToAction.text}
      </Link>
    </div>
  </section>
)

PricingTable.defaultProps = {
  id: null,
}

PricingTable.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node.isRequired,
  headline: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  imgFluid: PropTypes.object.isRequired,
  callToAction: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
  }),
}

export default PricingTable
