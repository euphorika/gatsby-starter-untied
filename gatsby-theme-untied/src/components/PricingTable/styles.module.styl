@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

:root
  --section-highlight-color highlight-color

.pricing-table-section
  margin 0 auto
  max-width 300px
  padding 15px
  background-color var(--section-highlight-color)
