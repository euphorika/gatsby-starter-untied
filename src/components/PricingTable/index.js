import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'

import styles from './styles.module.styl'

const PricingTable = ({ children, headline, price, callToAction }) => (
  <section className={styles.pricingTableSection}>
    <h3>{headline}</h3>
    <p>{price}</p>
    <div className={styles.description}>
      {children}
    </div>
    <div className={styles.callToAction}>
      <Link to={callToAction.link}>{callToAction.text}</Link>
    </div>
  </section>
)

PricingTable.propTypes = {
  children: PropTypes.node.isRequired,
  headline: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  callToAction: PropTypes.shape({
    text: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired
  })
}

export default PricingTable
