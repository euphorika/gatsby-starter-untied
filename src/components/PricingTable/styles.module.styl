@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.pricing-table-section
  margin 0
