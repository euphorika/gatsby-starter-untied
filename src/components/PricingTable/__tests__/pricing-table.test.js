import React from 'react'
import renderer from 'react-test-renderer'
import PricingTable from '../'

describe('PricingTable', () => {
  it('renders correctly', () => {
    const callToAction = {
      text: 'Text',
      link: '/',
    }
    const pricingTable = (
      <PricingTable
        headline="headline"
        price="1.22"
        callToAction={callToAction}
      >
        Description
      </PricingTable>
    )

    const tree = renderer.create(pricingTable).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
