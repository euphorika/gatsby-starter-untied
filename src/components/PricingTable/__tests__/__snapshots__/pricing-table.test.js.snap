// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingTable renders correctly 1`] = `
<section
  className="pricingTableSection"
>
  <h3>
    headline
  </h3>
  <p>
    1.22
  </p>
  <div
    className="description"
  >
    Description
  </div>
  <div
    className="callToAction"
  >
    <Link
      to="/"
    >
      Text
    </Link>
  </div>
</section>
`;
